FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,20 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 	
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-        
+        self.client.get(from: self.url) { [weak self] result in
+            guard self != nil else { return }
+            switch result {
+            case .failure(_):
+                completion(.failure(Error.connectivity))
+                break
+            case .success((let data, let response)):
+                guard response.statusCode == 200,
+                      let _ = try? JSONSerialization.jsonObject(with: data) else {
+                    return completion(.failure(Error.invalidData))
+                }
+                
+                return completion(FeedImageMapper.map(from: data, response: response))
```

Comment:
Looks like this logic is duplicated - it's doing the same in the `FeedImageMapper`. So you could remove it.

```suggestion
                completion(FeedImageMapper.map(from: data, response: response))
```